File: C/unit2/brute_force/string_match.c

#include <stdio.h>
#include <string.h>

int find(char *text, char *pat)
{
    int n = strlen(text);
    int m = strlen(pat);
    for (int i = 0; i <= n - m; i++)
    {
        int j = 0;
        while (j < m && text[i + j] == pat[j])
        {
            j++;
        }
        if (j == m)
        {
            return i;
        }
    }
    return -1;
}

int main()
{
    char txt[] = "hi hello namskara";
    char pat[] = "helloz";
    int reas = find(txt, pat);
    if (reas != -1)
    {
        printf("Pattern found at index %d", reas);
    }
    else
    {
        printf("pattern not in the text");
    }
}

File: Java/unit2/brute_force/string_match.java
package Java.unit2.brute_force;

public class string_match {
    public static void search(String txt, String pat) {
        int M = pat.length();
        int N = txt.length();

        for (int i = 0; i <= N - M; i++) {

            int j;

            for (j = 0; j < M; j++)
                if (txt.charAt(i + j) != pat.charAt(j))
                    break;

            if (j == M)
                System.out.println("Pattern found at index " + i);

        }
    }

    public static void main(String[] args) {
        String txt = "AABAACAADAABAAABAA";
        String pat = "AABA";
        search(txt, pat);
    }
}
